src/main/java/org/skife/jdbi/v2/ClasspathStatementLocator.java

```diff
@@ -74,20 +75,32 @@ public static boolean looksLikeSql(String sql)
     @SuppressFBWarnings("DM_STRING_CTOR")
     public String locate(String name, StatementContext ctx)
     {
-        final String cache_key;
+        final Object cache_key;
+        boolean isSqlObjectMethod = false;
         if (ctx.getSqlObjectType() != null) {
-            cache_key = '/' + mungify(ctx.getSqlObjectType().getName() + '.' + name) + ".sql";
+            if (ctx.getSqlObjectMethod() != null) {
```

Comment:
Is there ever a case where `sqlObjectType` is set, but `sqlObjectMethod` is not?